docs: update daemon documentation (#3227)
canonical/ubuntu-pro-client: docs/explanations/what_is_the_daemon.rst

What is the Pro daemon?
*******************************

Ubuntu Pro Client sets up a daemon for two types of machines:

* Azure and GCP generic cloud images
* Cloud Pro images

For the first type of images, the daemon is responsible

> to detect if an Ubuntu Pro license has been purchased for the machine.

Reviewer: ```suggestion
for detecting if an Ubuntu Pro license has been purchased for the machine.
```

If a Pro license is detected, then the machine is automatically attached.

For the Pro cloud images, the daemon is responsible for retrying the auto-attach
operation if it fails on first boot. When that happens, users will be notified in

> MOTD or `pro status` with a message with pattern:

Reviewer: ```suggestion
MOTD or when they run `pro status` a message like:
```

.. code-block:: text

   Failed to automatically attach to an Ubuntu Pro subscription 1 time
   The failure was due to: ERROR_MSG
   The next attempt is scheduled for RETRY_DATE
   You can try manually with `sudo pro auto-attach`.

> The retry auto-attach operation will keep retrying for a **month**. If at the end, we cannot attach

Reviewer: ```suggestion
The "retry auto-attach" operation will keep retrying for a **month**. If at the end, we cannot attach
```

Reviewer: For a month, but at what frequency?

Author: It changes as time goes by. It starts retrying after 15min after a failure and every retry takes longer than the last attempt. Do you think it is worth explaining that ?

Reviewer: I think probably not for now. The explanation might lead to more confusion or questions. Happy to leave it as is for now!
